html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,326 @@
+   *   List of documents keyed by IDs.
+   */
+  public function getDocuments(int $limit = 50): array {
+    $this->getLogger()->info('Retrieving documents from the WorkDay.');
+
+    $documents = [];
+
+    try {
+      $timeout = $this->settings['timeout'] ?? 10;
+      $base_url = $this->settings['base_url'] ?? '';
+      $tenant = $this->settings['tenant'] ?? '';
+      $url = $base_url . '/ccx/api/recruiting/v4/' . $tenant . '/jobPostings';
+
+      $auth = $this->getAuthToken();
+
+      $response = $this->httpClient->request('GET', $url, [
+        'connect_timeout' => $timeout,
+        'timeout' => $timeout,
+        'headers' => [
+          'Content-Type' => 'application/json',
+          'Accept' => 'application/json',
+          'Authorization' => 'Bearer ' . $auth,
+        ],
+      ]);
+
+      if ($response->getStatusCode() !== 200) {
+        throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+      }
+
+      $jobs = json_decode($response->getBody()->getContents(), TRUE);
+      foreach ($jobs['data'] as $job) {
+        // Skip jobs without title or URL.
+        if (empty($job['title']) || empty($job['url'])) {
+          continue;
+        }
+
+        $documents[] = [
+          'id' => $job['id'],
+          'field_job_closing_date' => $job['startDate'] ?? '',
+          'job_type' => $job['jobType']['descriptor'] ?? '',
+          'company' => $job['company']['descriptor'] ?? '',
+          'time_type' => $job['timeType']['descriptor'] ?? '',
+          'field_city' => $job['primaryLocation']['descriptor'] ?? '',
+          'field_country' => [$job['primaryLocation']['country']['descriptor'] ?? ''],
+          'url' => $job['url'] ?? '',
+          'title' => $job['title'],
+          'body' => $job['jobDescription'] ?? '',
+        ];
+      }
+
+      return $documents;
+    }
+    catch (\Exception $e) {
+      $this->getLogger()->error('Unable to retrieve the WorkDay documents: ' . $e->getMessage());
+      return [];
+    }
+  }
+
+  /**
+   * Process the WorkDay documents data.
+   *
+   * @param array $documents
+   *   The WorkDay documents data.
+   */
+  public function importDocuments(array $documents): void {
+    $source_id = $this->settings['source_id'] ?? 0;
+    $uid = $this->settings['uid'] ?? 2;
+
+    $errors = [];
+    $warnings = [];
+
+    foreach ($documents as $item) {
+      $name = $item['title'] ?? 'unknown';
+      try {
+        $guid = trim($item['url'] ?? '');
+        $this->url = $guid;
+
+        // Check if job already exist.
+        if ($this->jobExists($guid)) {
+          $this->getLogger()->notice(strtr('Updating job @guid', [
+            '@guid' => $guid,
+          ]));
+          $job = $this->loadJobByGuid($guid);
+          if (empty($job)) {
+            throw new ReliefwebImportExceptionViolation(strtr('Unable to load job @guid', [
+              '@guid' => $guid,
+            ]));
+          }
+          $this->updateJob($job, $item);
```

Comment:
`updateJob`'s code expect $item to be an object, so it should be cast to object here (or in the updateJob method).

Reply:
Fixed